### PR TITLE
redefined finalize() in Family, Human, and Pet classes for removal messages. Added loops in Main to create Human objects from 10,000 to 10 million. Switched to AutoCloseable due to finalize() deprecation.

### PR DESCRIPTION
**Pull request by hikmataliyev059**

### CODE REVIEW

File: src/main/java/hw01/Human.java [hunk 1]
```diff
@@ -113,33 +112,30 @@ public void describePet() {
...
-        return "Human{" +
-                "name='" + name + '\'' +
-                ", surname='" + surname + '\'' +
-                ", dateOfBirthYear=" + dateOfBirthYear +
-                ", iq=" + iq +
-                ", pet=" + pet +
-                ", family=" + family +
-                ", schedule=" + Arrays.toString(schedule) +
-                '}';
+        return String.format("{name='%s', surname='%s', dateOfBirthYear=%d, iq=%d, pet=%s, family=%s, schedule=%s}", name, surname, dateOfBirthYear, iq, pet, family, Arrays.toString(schedule));
```
Comment (sabuhihuseynov) [inline src/main/java/hw01/Human.java:138]: should be Human in the begining

File: src/main/java/hw02/Family.java [hunk 2]
```diff
@@ -58,4 +55,10 @@ public Pet getPet() {
     public void setPet(Pet pet) {
         this.pet = pet;
     }
+
+    @Override
+    public void close() {
+        System.out.println("Family object is being removed.");
```
Comment (sabuhihuseynov) [inline src/main/java/hw02/Family.java:61]: Only implement AutoCloseable if the class manages external resources that need to be explicitly released. For a regular object like a Family instance, implementing AutoCloseable might not be meaningful.

**Review by sabuhihuseynov — CHANGES_REQUESTED** [review]
pls look

File: src/main/java/hw01/Human.java [hunk 3]
```diff
@@ -5,7 +5,7 @@
 import java.util.Arrays;
 import java.util.Random;
 
-public class Human {
+public class Human implements AutoCloseable {
```
[the remark this answers is not in the document]
Comment (hikmataliyev059) [reply src/main/java/hw01/Human.java:8]: because the finalize() method is declared deprecated and does not support it after java 9

**Pull request merged**